btk/measure.py

```diff
@@ -220,38 +262,49 @@ def run_batch(self, batch, index, **kwargs):
                     "contain an astropy table as the value of the key 'catalog'."
                 )
 
-            if isinstance(batch["blend_images"], np.ndarray):
-                if not (
-                    "x_peak" in out["catalog"].colnames and "y_peak" in out["catalog"].colnames
-                ):
-                    raise ValueError(
-                        "The output catalog of at least one of your measurement functions does not"
-                        "contain the 'x_peak' and 'y_peak' columns which are mandatory for a single"
-                        "survey study."
-                    )
-
-            if isinstance(batch["blend_images"], dict):
-                if not ("ra" in out["catalog"].colnames and "dec" in out["catalog"].colnames):
-                    raise ValueError(
-                        "The output catalog of at least one of your measurement functions does not"
-                        "contain the 'ra' and 'dec' columns which are mandatory for a"
-                        "multi-resolution study."
-                    )
+            if not ("ra" in out["catalog"].colnames and "dec" in out["catalog"].colnames):
+                raise ValueError(
+                    "The output catalog of at least one of your measurement functions does not"
+                    "contain the mandatory 'ra' and 'dec' columns"
+                )
 
             for key in ["deblended_images", "segmentation"]:
                 if key in out and out[key] is not None:
-                    if not isinstance(out[key], np.ndarray):
-                        raise TypeError(
-                            f"The output '{key}' of at least one of your measurement"
-                            f"functions is not a numpy array."
-                        )
-                    if key == "deblended_images":
-                        if not out[key].shape[-3:] == batch["blend_images"].shape[-3:]:
-                            raise ValueError(
-                                f"The shapes of the blended images in your {key} don't "
-                                f"match for at least one your measurement functions."
-                                f"{out[key].shape[-3:]} vs {batch['blend_images'].shape[-3:]}"
+                    if len(surveys) == 1:
+                        if not isinstance(out[key], np.ndarray):
+                            raise TypeError(
```

Comment:
Some of the error cases are not covered by tests (see below for codecov messages). Do you mind updating the tests for coverage? If you prefer to do it later (or for me to do it) feel free to just open an issue about it so we don't forget.

Reply:
I feel like this is not an urgent issue as long as we keep the coverage up, so I will open an issue for the time being.